File: strategy/typescript/glados/test/move/chiptime.lua
local ChipTimeTest = Class("Test.Move.ChipTimeTest", require "group/move/base")

local World = require "../base/world"
local Physics = require "observer/physics"
local Pass = require "task/shared/pass"

ChipTimeTest.MIN_ROBOTS = 1
ChipTimeTest.MAX_ROBOTS = 1

function ChipTimeTest.canStart()
	return true
end

function ChipTimeTest:_init()
	local startPos = World.Ball.pos:copy()
	self._endPos = Vector(0, 0)
	local timePredicted = Physics.chipPassTime(startPos, self._endPos)
	log("Time needed: ".. timePredicted)
end

function ChipTimeTest:_canContinue()
	return true
end

function ChipTimeTest:_updateTasks()
	local taskAssignments = {}

	taskAssignments[self._robots[1]] = { class = Pass,
		params = { nil, self._endPos, true, 0 } }
	return taskAssignments
end

return ChipTimeTest

File: strategy/typescript/glados/test/move/chiptime.ts
let ChipTimeTest = Class("Test.Move.ChipTimeTest", require "group/move/base")

import * as World from "base/world";
import * as Physics from "glados/observer/physics";
import {Pass} from "glados/task/shared/pass";

ChipTimeTest.MIN_ROBOTS = 1
ChipTimeTest.MAX_ROBOTS = 1

function ChipTimeTest.canStart () {
	return true
}

function ChipTimeTest:_init () {
	let startPos = World.Ball.pos
	this._endPos = new Vector(0, 0)
	let timePredicted = Physics.chipPassTime(startPos, this._endPos)
	log("Time needed: " +  timePredicted)
}

function ChipTimeTest:_canContinue () {
	return true
}

function ChipTimeTest:_updateTasks () {
	let taskAssignments = {}

	taskAssignments[this._robots[0]] = { class: Pass,
		params = [ undefined, this._endPos, true, 0 ] }
	return taskAssignments
}

return ChipTimeTest
